include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -547,40 +548,50 @@ struct __result_and_scratch_storage
     }
 
   public:
-    __result_and_scratch_storage(const _ExecutionPolicy& __exec_, ::std::size_t __scratch_n)
-        : __exec{__exec_}, __scratch_n{__scratch_n}, __use_USM_host{__use_USM_host_allocations(__exec.queue())},
-          __supports_USM_device{__use_USM_allocations(__exec.queue())}
+    __result_and_scratch_storage(const _ExecutionPolicy& __exec_, std::size_t __result_n, std::size_t __scratch_n)
```

Comment:
```suggestion
    __result_and_scratch_storage(const _ExecutionPolicy& __exec_, std::size_t __result_n = 0, std::size_t __scratch_n = 0)
```

Reply:
I think this change was discussed here: https://github.com/oneapi-src/oneDPL/pull/1770#discussion_r1705980892. I still think that explicitly stating the value is clearer and I would prefer to leave it as is.